options/options_test.cc

```diff
@@ -930,16 +930,13 @@ TEST_F(OptionsTest, GetBlockBasedTableOptionsFromString) {
             new_opt.cache_index_and_filter_blocks);
   ASSERT_EQ(table_opt.filter_policy, new_opt.filter_policy);
 
-  // unrecognized filter policy config
-  s = GetBlockBasedTableOptionsFromString(config_options, table_opt,
-                                          "cache_index_and_filter_blocks=1;"
```

Comment:
Why are we removing `"cache_index_and_filter_blocks=1;"` in the test?

Reply:
It seems irrelevant. The cases of

```
  ASSERT_EQ(table_opt.cache_index_and_filter_blocks,
            new_opt.cache_index_and_filter_blocks);
```

after

```
  ASSERT_NOK(GetBlockBasedTableOptionsFromString(...
```

seem especially useless because they only test that a bad assignment wasn't made. It does not distinguish no assignment from proper assignment before failure. I don't know what's intended.